frontend/src/tree/TreeItemContextMenu.tsx

```diff
@@ -44,60 +74,114 @@ export const TreeItemContextMenu = ({
     }
   };
 
+  const [deleteTreeItem, { loading: deleteTreeItemLoading, data: deleteTreeItemData, error: deleteTreeItemError }] =
+    useMutation<GQLDeleteTreeItemData, GQLDeleteTreeItemVariables>(deleteTreeItemMutation);
+
+  const deleteItem = () => {
+    const input: GQLDeleteTreeItemInput = {
+      id: uuid(),
+      editingContextId,
+      representationId: treeId,
+      treeItemId: item.id,
+    };
+    deleteTreeItem({ variables: { input } });
+    onClose;
+  };
+
+  const handleError = (loading, data, error) => {
+    if (!loading) {
+      if (error) {
+        openToast;
+      }
+      if (data) {
+        const keys = Object.keys(data);
+        if (keys.length > 0) {
+          const firstKey = keys[0];
+          const firstField = data[firstKey];
+          if (firstField.__typename === 'ErrorPayload') {
+            openToast;
+          }
+        }
+      }
+    }
+  };
+
+  useEffect(() => {
+    handleError(deleteTreeItemLoading, deleteTreeItemData, deleteTreeItemError);
+  }, [deleteTreeItemLoading, deleteTreeItemData, deleteTreeItemError, handleError]);
+
   return (
-    <Menu
-      id="treeitem-contextmenu"
-      anchorEl={menuAnchor}
-      keepMounted
-      open
-      onClose={onClose}
-      data-testid="treeitem-contextmenu"
-      disableRestoreFocus={true}
-      getContentAnchorEl={null}
-      anchorOrigin={{
-        vertical: 'bottom',
-        horizontal: 'right',
-      }}
-    >
-      {treeItemContextMenuContributions
-        .filter((contribution) => contribution.props.canHandle(item))
-        .map((contribution, index) => {
-          const props: TreeItemContextMenuComponentProps = {
-            editingContextId,
-            item,
-            readOnly,
-            onClose,
-            selection,
-            setSelection,
-            expandItem,
-            key: index.toString(),
-            treeId: treeId,
-          };
-          const element = React.createElement(contribution.props.component, props);
-          return element;
-        })}
-      {item.editable ? (
-        <MenuItem
-          key="rename"
-          onClick={enterEditingMode}
-          data-testid="rename-tree-item"
-          disabled={readOnly}
-          aria-disabled
-        >
-          <ListItemIcon>
-            <EditIcon fontSize="small" />
-          </ListItemIcon>
-          <ListItemText primary="Rename" />
-        </MenuItem>
-      ) : null}
-      {item.deletable ? (
-        <MenuItem key="delete-tree-item" onClick={deleteItem} data-testid="delete" disabled={readOnly} aria-disabled>
-          <ListItemIcon>
-            <DeleteIcon fontSize="small" />
-          </ListItemIcon>
-          <ListItemText primary="Delete" />
-        </MenuItem>
-      ) : null}
-    </Menu>
+    <div>
```

Comment:
Use `<> </>` instead of a div. You do not need an additional div which may change the layout